web_widget_date_interval/static/src/js/web_widget_date_interval.js

```diff
@@ -0,0 +1,155 @@
+                this.on('change:value', this, this.reinitialize);
+            },
+            _get_date: function () {
+                return instance.web.str_to_date(
+                    instance.web.parse_value(
+                        this.get_value(), this, Date.now()
+                    ).substring(0, 10)
+                );
+            },
+            weeknumber_iso_weeknumber: function (date) {
+                // GetISOWeek looks at the UTC time, but users expect walltime
+                var local = date.clone().clearTime().add(12).hours();
+                return parseInt(local.getISOWeek(), 10);
+            },
+            weeknumber_iso_year: function (date) {
+                // The ISO year of a week is the year of the week's thursday
+                var _date = date.clone();
+                if (!_date.is().monday()) {
+                    _date.last().monday();
+                }
+                return _date.next().thursday().getFullYear();
+            },
+            weeknumber_iso_get_years: function () {
+                if (this.options.hide_years) {
+                    return [];
+                }
+                var result = [],
+                    years_before = this.options.years_before,
+                    years_after = this.options.years_after,
+                    current_date = this._get_date();
+                for (
+                    var year=current_date.getFullYear() - years_before;
+                    year <= current_date.getFullYear() + years_after;
+                    year++
+                ) {
+                    result.push(year);
+                }
+                return result;
+            },
+            weeknumber_iso_get_weeks: function () {
+                var result = [],
+                    last_day = this._get_date(),
+                    min_week = 1,
+                    max_week = 52;
+                if (!last_day.is().december()) {
+                    last_day.december();
+                }
+                last_day.moveToLastDayOfMonth();
+                if (this.weeknumber_iso_weeknumber(last_day) > 1) {
+                    max_week = this.weeknumber_iso_weeknumber(last_day);
+                }
+                if (this.options.week_type === 'buttons') {
+                    min_week = Math.max(
+                        min_week,
+                        this.weeknumber_iso_weeknumber(this._get_date()) -
+                        this.options.weeks_before
+                    );
+                    max_week = Math.min(
+                        max_week,
+                        this.weeknumber_iso_weeknumber(this._get_date()) +
+                        this.options.weeks_after
+                    );
+                }
+                for (var week=min_week; week <= max_week; week++) {
+                    result.push(week);
+                }
+                return result;
+            },
+            weeknumber_iso_date_from_week: function (year, week) {
```

Comment:
A little bit confused here with the ISO references. What are the advantages of using that ISO calendar system?